### DISCUSSION TITLE
Debug logging extended

### DISCUSSION THREAD

**Discussion by CptArn** [gicamm/homeassistant-comelit#37] [General]
Trying to set up the integration and couldn't get the entities to load in home assistant. When enabling the debugging of the custom component, I see some things are happening but I can't follow what is going wrong. Can there be added more feedback or can you help out on where the issue may be? Added a screenshot of the logging.

![image](https://user-images.githubusercontent.com/1754536/223554730-6df03c27-533e-485d-9007-23bf2ea645de.png)

**Comment by gicamm** [comment]
can you share your configuration?
cheers
G

**Comment by CptArn** [comment]
comelit:
  hub:
    host: 127.0.0.1
    port: 1883
    mqtt-user: mqttuser
    mqtt-password: mqttpassword
    username: comelitUser (like in the app)
    password: comelitPassword (like in the app)
    serial: serial
    client: homeassistant
    scan_interval: 2

**Comment by gicamm** [comment]
A couple of considerations.
The 'host' is the HUB IP address. Do you have some forwarding? Otherwise, you need to specify the right address.
The serial must contain only digits. (i.e. 'hsrv-0123456789' -> '0123456789' )

**Comment by CptArn** [comment]
I see here that the mqtt connection is happening against the hub host? https://github.com/gicamm/homeassistant-comelit/blob/c066633918bf951564717dd430a0f844fa68e5bb/custom_components/comelit/hub.py#L226 
I have no forwarding but host the mqtt server locally.
The serial only contains the numbers indeed.

**Comment by gicamm** [comment]
The integration must reach the comelit hub, not the local mqtt server.
You must specify the comelit hub IP address.

**Comment by CptArn** [comment]
Where do I find the mqtt username/password for the comelit hub? I'm not completely following how the config works.

**Comment by CptArn** [comment]
Mhhh, the hub isn't the same as the icona manager?

**Comment by gicamm** [comment]
You need clarification, and I am trying to understand why you don't notice what I wrote.
To make the integration work, you will need to specify the comelit HUB IP address.
127.0.0.1 is your local MQTT server address. But how the home assistant should talk with the hub? There is no magic in home assistant.
You must replace 127.0.0.1 with the hub address. The HUB is connected to your home LAN network. You must specify that address in the 'host' configuration field.
Are you using the comelit APP? If yes you can use the same username and password.

**Reply by CptArn** [reply]
I'm following that, but I'm guessing the issue resides in the fact that the icona manager doesn't expose an mqtt server. I'm following that the hub is hosting an MQTT server. I'm not sure where to find the login credentials of the hub (if the icona manager even hosts one).

**Comment by CptArn** [comment]
I'm following that, but I'm guessing the issue resides in the fact that the icona manager doesn't expose an mqtt server. I'm following that the hub is hosting an MQTT server. I'm not sure where to find the login credentials of the hub (if the icona manager even hosts one).

**Reply by gicamm** [reply]
the icona manager is not an mqtt server. The comelit hub supports mqtt. Who set up your phone?
